index.d.ts

```diff
@@ -2910,23 +2910,23 @@ declare namespace Eris {
         members?: Collection<Omit<Member, "user" | "deaf" | "mute">>;
         roles?: Collection<Role>;
         channels?: Collection<PartialChannel>;
-        messages?: Collection<Message>;
+        messages?: Collection<Message<T>>;
       };
       options?: InteractionDataOptions[];
     };
-    guildID?: string;
-    member?: Member;
+    guildID: T extends GuildTextableWithThread ? string : undefined;
+    member: T extends GuildTextableWithThread ? Member : null;
```

Comment:
Member is not nullable, and these types set the guildID as the explicit value "undefined", which it is not - never or something would likely be more accurate